lib/server.js

```diff
@@ -891,13 +891,15 @@ Server.prototype._run = function _run(req, res, route, chain, cb) {
                 return (next());
             }
 
+            handlerName = (chain[i].handlerName ||
```

Comment:
I just pushed some docs changes to add handler naming possibility, tell me if this is not the way you would see it.